airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -1227,6 +1256,45 @@ def execute(self, context: Context) -> str | None:
 
         return self.job_id
 
+    def start_job_run_after_defer(self, context: Context, event: dict[str, Any] | None = None) -> None:
```

Comment:
I've seen in the docs that you can do a re-entering execute if you add `event` as a parameter, I'm wondering if it would make sense here.
I'd save some code duplication, with very little `if else` needed I think (since you can rely on the `if app_state not in EmrServerlessHook.APPLICATION_SUCCESS_STATES:` that's already there for the code to take the correct path.
Just a little check on `event["status"] == "success"` at the top would suffice.

Reply:
That's a really good point, I totally missed that in the documentation. It turns out I didn't need to check for `event["status"]` at all, because there is no need for it. The code follows the correct path because of this statement `if app_state not in EmrServerlessHook.APPLICATION_SUCCESS_STATES:`